snapcraft/pluginhandler.py

```diff
@@ -196,8 +196,18 @@ def is_dirty(self, step):
             # about, and we're comparing it to those same keys in the current
             # YAML (taken from self.code.options). If they've changed, then
             # this step is dirty and needs to run again.
-            return state.properties != state.properties_of_interest(
-                self.code.options)
+            if state.properties != state.properties_of_interest(
+                    self.code.options):
+                return True
+
+        with contextlib.suppress(AttributeError):
+            # state.project_options contains the old project options that this
+            # step cares about, and we're comparing it to those same options in
+            # the current project. If they've changed, then this step is dirty
+            # and needs to run again.
+            if state.project_options != state.project_options_of_interest(
```

Comment:
I previously had that on line 199 and I had to change it in order to add this section. I figured I'd just make them consistent and easy to expand.